.circleci/config.yml

```diff
@@ -0,0 +1,59 @@
+version: 2.1
+
+references:
+  images:
+    go: &GOLANG_IMAGE circleci/golang:latest
+  environments:
+    tmp: &TEST_RESULTS_PATH /tmp/test-results # path to where test results are saved
+
+# reusable 'executor' object for jobs
+executors:
+  go:
+    docker:
+      - image: *GOLANG_IMAGE
+    environment:
+      - TEST_RESULTS: *TEST_RESULTS_PATH
+
+jobs:
+  go-fmt-and-test:
+    executor: go
+    steps:
+      - checkout
+      - run: mkdir -p $TEST_RESULTS
+
+      # Restore go module cache if there is one
+      - restore_cache:
+          keys:
+            - go-getter-modcache-v1-{{ checksum "go.mod" }}
```

Comment:
It would be good to add another key after the first to still fetch the most recent cache if the `go.mod` file changes. Something like 
```yaml
- restore_cache:
  keys:
    - go-getter-modcache-v1-{{ checksum "go.mod" }}
    - go-getter-modcache-v1-
```

[For reference](https://circleci.com/docs/2.0/caching/)